src/containers/FilmFrontpage/FilmFrontpage.jsx

```diff
@@ -0,0 +1,251 @@
+import { injectT } from '@ndla/i18n';
+import {
+  FilmSlideshow,
+  MovieGrid,
+  AboutNdlaFilm,
+  FilmMovieSearch,
+  FilmMovieList,
+} from '@ndla/ui';
+
+const ARIA_FILMCATEGORY_ID = 'movieCategoriesId';
+
+class FilmFrontpage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      resourceTypeSelected: null,
+      columnWidth: 260,
+      columnsPrSlide: 1,
+      margin: 26,
+    };
+    this.onChangeResourceType = this.onChangeResourceType.bind(this);
+    this.setScreenSize = this.setScreenSize.bind(this);
+    this.setScreenSizeDebounced = debounce(() => this.setScreenSize(false), 50);
+    this.movieListRef = React.createRef();
+  }
+
+  componentDidMount() {
+    this.setScreenSize();
+    window.addEventListener('resize', this.setScreenSizeDebounced);
+  }
+
+  componentWillUnmount() {
+    this.setScreenSizeDebounced.cancel();
+    window.removeEventListener('resize', this.setScreenSizeDebounced);
+  }
+
+  onChangeResourceType(resourceTypeSelected) {
+    const loadingPlaceholderHeight = `${
+      this.movieListRef.current.getBoundingClientRect().height
+    }px`;
+
+    if (resourceTypeSelected) {
+      this.props.onSelectedMovieByType(resourceTypeSelected);
+    }
+
+    this.setState({
+      resourceTypeSelected,
+      loadingPlaceholderHeight,
+    });
+  }
+
+  setScreenSize() {
+    const screenWidth =
+      window.innerWidth || document.documentElement.clientWidth;
+
+    const currentBreakpoint = getCurrentBreakpoint();
+    let margin;
+    let itemSize;
+    if (screenWidth < 385) {
+      margin = 26;
+      itemSize = 130;
+    } else if (screenWidth < 450) {
+      margin = 26;
+      itemSize = 160;
+    } else if (currentBreakpoint === breakpoints.mobile) {
+      margin = 26;
+      itemSize = 200;
+    } else if (currentBreakpoint === breakpoints.tablet) {
+      margin = 52;
+      itemSize = 220;
+    } else if (currentBreakpoint === breakpoints.desktop) {
+      margin = 78;
+      itemSize = 240;
+    } else if (screenWidth < 1600) {
+      margin = 104;
+      itemSize = 260;
+    } else {
+      margin = 104;
+      itemSize = 300;
+    }
+
+    const columnsPrSlide = Math.floor((screenWidth - margin * 2) / itemSize);
+
+    /* eslint react/no-did-mount-set-state: 0 */
+    this.setState({
+      columnWidth: (screenWidth - margin * 2) / columnsPrSlide,
+      columnsPrSlide,
+      margin,
+    });
+    /* eslint react/no-did-mount-set-state: 1 */
+  }
+
+  render() {
+    const {
+      highlighted,
+      themes,
+      resourceTypes,
+      topics,
+      aboutNDLAVideo,
+      moviesByType,
+      fetchingMoviesByType,
+      moreAboutNdlaFilm,
+      language,
+      t,
+    } = this.props;
+    const { resourceTypeSelected, loadingPlaceholderHeight } = this.state;
+
+    const resourceTypeName =
+      resourceTypeSelected &&
+      resourceTypes.find(
+        resourceType => resourceType.id === resourceTypeSelected,
+      );
+
+    return (
```

Comment:
Jeg lurer på om vi kanskje skal prøve å splitte opp denne i mindre komponenter, for å gjøre det lettere å forstå den?